target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-glinet.dtsi

```diff
@@ -0,0 +1,339 @@
+			bias-pull-up;
+			output-high;
+		};
+	};
+
+	leds_pins: leds_pins {
+		white {
+			pins = "gpio35";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		blue {
+			pins = "gpio37";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
+
+	i2c_1_pins: i2c_1_pins {
+		mux {
+			pins = "gpio42", "gpio43";
+			function = "blsp2_i2c";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
+};
+
+&blsp1_uart3 {
+	pinctrl-0 = <&serial_3_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&blsp1_spi1 {
+	pinctrl-0 = <&spi_1_pins>;
+	pinctrl-names = "default";
+	cs-select = <0>;
+	status = "okay";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0>;
+		compatible = "n25q128a11";
+		linux,modalias = "m25p80", "n25q128a11";
+		spi-max-frequency = <50000000>;
+		use-default-sizes;
+	};
+};
+
+&blsp1_i2c3 {
+	pinctrl-0 = <&i2c_1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&prng {
+	status = "okay";
+};
+
+&cryptobam {
+	status = "okay";
+};
+
+&crypto {
+	status = "okay";
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qusb_phy_0 {
+	status = "okay";
+};
+
+&qusb_phy_1 {
+	status = "okay";
+};
+
+&ssphy_0 {
+	status = "okay";
+};
+
+&usb3 {
+	pinctrl-0 = <&usb_power_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&usb2 {
+	status = "okay";
+};
+
+&edma {
+	status = "okay";
+};
+
+&rpm {
+	status = "disabled";
+};
+
+&mdio {
+	status = "okay";
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+	reset-gpios = <&tlmm 74 GPIO_ACTIVE_LOW>;
+
+	ethernet_0: ethernet-phy-package@0 {
+		compatible = "qcom,qca8075-package";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+		qcom,package-mode = "psgmii";
+
+		qca8075_0: ethernet-phy@0 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <0>;
+		};
+
+		qca8075_1: ethernet-phy@1 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <1>;
+		};
+
+		qca8075_2: ethernet-phy@2 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <2>;
+		};
+	};
+};
+
+&switch {
+	status = "okay";
+
+	switch_lan_bmp = <(ESS_PORT2|ESS_PORT3)>; /* lan port bitmap */
+	switch_wan_bmp = <ESS_PORT1>; /* wan port bitmap */
+
+	qcom,port_phyinfo {
+		port@0 {
+			port_id = <1>;
+			phy_address = <0>;
+		};
+		port@1 {
+			port_id = <2>;
+			phy_address = <1>;
+		};
+		port@2 {
+			port_id = <3>;
+			phy_address = <2>;
+		};
+	};
+};
+
+&dp1{
+	phy-handle = <&qca8075_0>;
+	status = "okay";
+};
+
+&dp2{
+	phy-handle = <&qca8075_1>;
+	status = "okay";
+};
+
+&dp3{
+	phy-handle = <&qca8075_2>;
+	status = "okay";
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qpic_nand {
+	status = "okay";
+
+	nand@0 {
+		reg = <0>;
+
+		nand-ecc-strength = <4>;
+		nand-ecc-step-size = <512>;
+		nand-bus-width = <8>;
+
+        partitions {
+			compatible = "qcom,smem-part";
+		};
+	};
+};
+
+&wifi {
+	status = "okay";
+	qcom,ath11k-fw-memory-mode = <1>;
+ };
+
+&CPU0 {
+	/delete-property/ cpu-supply;
```

Comment:
How does CPU scaling work with this?

Reply:
I get that, but you are completely disabling CPUFreq with this.
What you need is to disable the RPM regulator node, and add a fixed regulator that you will put in the OPP table instead of removing the supply

Reply:
I don't quite understand how to do it, can you give me an example? Like [this](https://github.com/robimarko/linux/commit/41bf6773b9dd4c18a2d73624d0cd25477ede2aec#diff-86818f2f2641ea3f0f018ea61ef4e34af2619e186bd986adaeaa33b4d2143f69R113)?

Reply:
You need to define a fixed regulator like so:

` {
	ipq60xx_cpu_regulator: cpu-regulator {
		compatible = "regulator-fixed";
		regulator-name = "ipq60xx-cpu-regulator";
		regulator-min-microvolt = <1100000>;
		regulator-max-microvolt = <1100000>;
		regulator-always-on;
		regulator-boot-on;
	};
};

`

And use that instead of the the apc_vreg cpu-supply

Reply:
@kirdesde Thanks for your reply, following your suggestion did not make cpufreq work, maybe there is something wrong with my operation. But cupfreq works fine when I remove the cpu-supply node.
`[    1.119221] cpufreq: cpufreq_online: CPU0: Running at unlisted initial frequency: 799999 KHz, changing to: 864000 KHz`

Reply:
You still haven't fixed the cpu-supply topic.

Please add the fixed cpu regulator and disable the rpm node 

`/delete-node/ rpm_requests;`

should work.

Please report back and post the entire kernel log.